networkx/algorithms/isomorphism/vf2pp.py

```diff
@@ -97,6 +98,13 @@ def precheck(G1, G2, G1_labels, G2_labels):
     if sorted(d for n, d in G1.degree()) != sorted(d for n, d in G2.degree()):
         return False
 
+    if not node_labels:
+        G1_labels.update({node: "green" for node in G1.nodes()})
+        G2_labels.update({node: "green" for node in G2.nodes()})
+    else:
+        G1_labels.update(G1.nodes(data=node_labels))
+        G2_labels.update(G2.nodes(data=node_labels))
+
```

Comment:
The reason i put this in here and not before is to avoid useless computations, in case the first two non-label checks fail.